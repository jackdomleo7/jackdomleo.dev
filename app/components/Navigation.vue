<template>
  <nav class="nav" :class="{ 'nav--sticky': !navIsAtTop || isMobileNavOpen }">
    <div class="nav__wrapper">
      <nuxt-link to="/" class="nav__name">
        Jack Domleo
        <span class="sr-only"> - Home</span>
      </nuxt-link>
      <nuxt-link to="/" class="nav__logo">
        <Icon name="custom:j-icon" mode="svg" />
        <span class="sr-only">Home</span>
      </nuxt-link>
      <div v-if="!isMobile" class="nav__primary">
        <ul class="nav__primary-list">
          <li v-for="navItem in navItems" :key="navItem.text" class="nav__item">
            <nuxt-link v-if="navItem.url" :to="navItem.url">
              <span>{{ navItem.text }}</span>
            </nuxt-link>
            <div v-else tabindex="0">
              <span>{{ navItem.text }}</span>
              <Icon class="nav__item-chevron" name="custom:chevron-down" mode="svg" />
            </div>
            <ul v-if="navItem.submenu">
              <li v-for="subItem in navItem.submenu" :key="subItem.text">
                <nuxt-link v-if="subItem.url" :to="subItem.url">
                  {{ subItem.text }}
                </nuxt-link>
              </li>
            </ul>
          </li>
        </ul>
      </div>
      <button
        v-if="isMobile"
        class="nav__hamburger"
        :class="{ 'nav__hamburger--open': isMobileNavOpen }"
        @click="isMobileNavOpen = !isMobileNavOpen"
      >
        <div class="nav__hamburger-icon">
          <span></span>
          <span></span>
          <span></span>
          <span></span>
        </div>
        <span class="sr-only">{{ isMobileNavOpen ? 'Close' : 'Menu' }}</span>
      </button>
    </div>
    <ul v-if="isMobile" class="nav__more" :class="{'nav__more--open': isMobile && isMobileNavOpen}">
      <li v-for="navItem in getMobileNavItems()" :key="navItem.text">
        <nuxt-link :to="navItem.url!" @click="isMobileNavOpen = false">
          {{ navItem.text }}
        </nuxt-link>
      </li>
    </ul>
  </nav>
</template>

<script lang="ts" setup>
import type { FixedLengthArray } from 'type-fest';

const route = useRoute()

interface INav {
  text: string;
  url: string|null;
  submenu?: {
    text: string;
    url: string;
  }[]
}

const isMobile = ref(false)
const isTouchscreen = ref(false)
const isMobileNavOpen = ref(false)
const navIsAtTop = ref(false)

const navItems: FixedLengthArray<INav, 3> = [ // No more than 3 top-level items
  {
    text: 'Projects',
    url: '/projects'
  },
  {
    text: 'Work',
    url: '/work'
  },
  {
    text: 'More',
    url: null,
    submenu: [
      {
        text: 'Blog',
        url: '/blog'
      },
      {
        text: 'Links',
        url: '/links'
      }
    ]
  }
]

function getMobileNavItems(): Omit<INav, 'submenu'>[] {
  const items: Omit<INav, 'submenu'>[] = []

  navItems.forEach(item => {
    if (!item.submenu) {
      items.push(item)
    }
    else {
      item.submenu.forEach(subItem => {
        items.push(subItem)
      })
    }
  })

  return items.filter(x => x.url)
}

watch(route, () => {
  isMobileNavOpen.value = false
})

onMounted(() => {
  setResponsiveness()
  handleScroll()
  window.addEventListener('resize', setResponsiveness)
  window.addEventListener('scroll', handleScroll)
})

onUnmounted(() => {
  window.removeEventListener('resize', setResponsiveness)
  window.removeEventListener('scroll', handleScroll)
})

function setResponsiveness (): void {
  const navBreak = window.getComputedStyle(document.querySelector('nav.nav')!).getPropertyValue('--nav-break')
  isMobile.value = !window.matchMedia(`(min-width: ${navBreak})`).matches
  isTouchscreen.value = !window.matchMedia('(hover: hover)').matches

  if (window.matchMedia(`(min-width: ${navBreak})`).matches) {
    isMobileNavOpen.value = false
  }
}

function handleScroll (): void {
  navIsAtTop.value = !(document.body.scrollTop > 10 || document.documentElement.scrollTop > 10)
}
</script>

<style lang="scss" scoped>
$navBreak: $responsive-large-tablet;

.nav {
  --nav-break: #{$navBreak};
  position: fixed;
  z-index: 999;
  width: 100vw;
  background-color: var(--color-bg);
  color: var(--color-primary);
  z-index: 99;
  position: fixed;
  height: 4rem;
  transition: 360ms ease;
  transition-property: box-shadow, background-color;
  will-change: box-shadow, background-color;
  box-shadow: 0 0 0.5rem transparent;

  &--sticky {
    box-shadow: var(--shadow);
    background-color: var(--color-fg1);

    .nav__logo {
      top: 50%;
      transform: translate(-50%, -50%);
      width: 3.5rem;
    }
  }

  &__wrapper {
    display: flex;
    align-items: stretch;
    gap: 1rem;
    justify-content: space-between;
    padding: 0.25rem 2rem;
    height: 100%;
    position: relative;
    z-index: 1;
    background-color: inherit;

    @media (min-width: $navBreak) {
      max-width: 104rem;
      margin-inline: auto;
      padding-inline: 10rem;
    }
  }

  &__name {
    color: var(--color-primary) !important;
    font-size: var(--text-body);
    text-decoration: none;
    cursor: pointer;
    display: inline-flex;
    align-items: center;

    @media (min-width: $responsive-large-mobile) {
      font-size: var(--text-large);
    }
  }

  &__logo {
    width: 5rem;
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
    top: 80%;
    transition-property: top, transform, width;
    transition: 360ms ease;
    will-change: top, transform, width;
  }

  &__primary {
    display: flex;

    &-list {
      list-style-type: none;
      padding-left: 0;
      margin: 0;
      padding-block: 0.25rem;
    }
  }

  &__item {
    display: inline-flex;
    font-size: var(--text-large);
    height: 100%;
    padding-inline: 0.25rem;

    @media (forced-colors: active) {
      border: 1px solid transparent;
    }

    &:hover,
    &:focus-within {
      ul {
        opacity: 1;
        visibility: visible;
      }
    }

    &-chevron {
      height: 1rem;
      width: 1rem;
      display: grid;
      place-items: center;
    }

    > div {
      cursor: default;
    }

    > a,
    > div {
      display: inline-flex;
      align-items: center;
      justify-content: center;
      gap: 0.25rem;
      height: 100%;
      width: 100%;
      padding-block: 0.25rem;
      min-width: 4rem;
      text-align: center;
      text-decoration: none;
      color: var(--color-primary);
      border-top: 1px solid transparent;
      border-bottom: 1px solid transparent;
      will-change: color, border-color;
      transition-property: color, border-color;
      transition: 160ms ease;

      &:hover,
      &:focus {
        border-bottom-color: var(--color-accent);

        + ul {
          opacity: 1;
          visibility: visible;
        }
      }
    }

    ul {
      opacity: 0;
      visibility: hidden;
      will-change: opacity, visibility;
      transition-property: opacity, visibility;
      transition: 160ms ease;
      position: absolute;
      top: 100%;
      margin: 0;
      list-style-type: none;
      background-color: var(--color-bg);
      box-shadow: var(--shadow);
      padding: 0.5rem 1.5rem;
      display: flex;
      flex-direction: column;
      gap: 0.5rem;
      font-size: var(--text-body);
      
      li {
        width: 100%;
      }

      a {
        padding: 0.5rem;
        display: flex;
        align-items: center;
        justify-content: flex-start;
        padding-block: 0.25rem;
        color: var(--color-primary);
        text-decoration: none;

        &:hover {
          text-decoration: underline;
          text-decoration-color: var(--color-accent);
          text-decoration-thickness: 2px;
        }
      }
    }
  }

  &__hamburger {
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 3.375rem;
    height: 1.9375rem;
    border: none;
    margin: 1rem 0;
    background-color: transparent;

    @media (forced-colors: active) {
      border: 1px solid ButtonText;
    }

    &-icon {
      width: 1.3rem;
      height: 1.1rem;
      position: relative;
      z-index: 2;
      transform: rotate(0deg);
      cursor: pointer;

      span {
        display: block;
        position: absolute;
        height: 0.125rem;
        width: 100%;
        background-color: var(--color-primary);
        border-radius: var(--border-radius-standard);
        opacity: 1;
        left: 0;
        transform: rotate(0deg);
        transition: 0.25s ease-in-out;

        @media (forced-colors: active) {
          background-color: ButtonText;
        }

        &:nth-child(1) {
          top: 0;
        }

        &:nth-child(2),
        &:nth-child(3) {
          top: 45%;
        }

        &:nth-child(4) {
          top: 90%;
        }

        .nav__hamburger--open & {
          &:nth-child(1),
          &:nth-child(4) {
            top: 45%;
            width: 0;
            left: 50%;
          }

          &:nth-child(2) {
            transform: rotate(45deg);
          }

          &:nth-child(3) {
            transform: rotate(-45deg);
          }
        }
      }
    }
  }

  &__more {
    position: absolute;
    top: 100%;
    left: 50%;
    transform: translateX(-50%);
    height: 0;
    background-color: var(--color-bg);
    border-top: 0 solid var(--color-accent);
    box-shadow: var(--shadow);
    padding: 0 3rem;
    margin: 0;
    min-width: 18rem;
    max-height: 24rem;
    list-style-type: none;
    transition-property: height, padding, border;
    transition: 260ms ease;
    will-change: height, padding, border;
    overflow: hidden;
    visibility: hidden;

    @media (forced-colors: active) {
      border: 1px solid transparent;
    }

    &--open {
      height: max-content;
      padding-block: 1rem;
      border-top-width: 4px;
      visibility: visible;
      overflow-y: auto;
      overscroll-behavior: contain;
    }

    li {
      &:not(:last-of-type) {
        border-bottom: 1px solid var(--color-fg1);
      }

      @media (forced-colors: active) {
        border: 1px solid transparent;
      }
    }

    a {
      text-decoration: none;
      display: flex;
      padding: 0.5rem;
      color: var(--color-primary);
    }
  }

  a.nuxt-link-active {
    color: var(--color-accent);
  }
}
</style>