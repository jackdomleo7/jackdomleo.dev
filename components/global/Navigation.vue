<template>
  <nav class="nav">
    <div class="nav__wrapper">
      <nuxt-link to="/" class="nav__name">
        Jack Domleo
        <span class="sr-only"> - Home</span>
      </nuxt-link>
      <nuxt-img src="~assets/images/j-icon.svg" alt="" class="nav__logo" />
      <div class="nav__primary">
        <ul class="nav__primary-list">
          <li v-for="(navItem, index) in navItems" :key="navItem.text" class="nav__item" :hidden="(isMobile && index === navItems.length - 1) || (!isMobile && index === 0)">
            <nuxt-link v-if="navItem.url" :to="navItem.url" :class="{'is-home': navItem.text.toLowerCase() === 'home'}">
              <svg-icon v-if="navItem.icon" :name="navItem.icon" />
              <span>{{ navItem.text }}</span>
            </nuxt-link>
            <a v-else href="javascript:void(0);">
              <span>{{ navItem.text }}</span>
              <svg-icon class="nav__item-chevron" name="chevron-down" />
            </a>
            <ul v-if="navItem.submenu">
              <li v-for="subItem in navItem.submenu" :key="subItem.text">
                <nuxt-link v-if="subItem.url" :to="subItem.url">
                  {{ subItem.text }}
                </nuxt-link>
              </li>
              <li v-if="navItem.text === 'More'" role="presentation">
                <theme-switcher id="desktop" />
              </li>
            </ul>
          </li>
        </ul>
        <button v-show="isMobile" class="nav__hamburger" :class="{'nav__hamburger--open': showMobileNav}" @click="showMobileNav = !showMobileNav">
          <span></span>
          <div class="nav__hamburger-icon">
            <span></span>
            <span></span>
            <span></span>
            <span></span>
          </div>
          <span class="sr-only">Show {{ showMobileNav ? 'more' : 'less' }}</span>
        </button>
      </div>
      <ul class="nav__more" :class="{'nav__more--mobile-open': isMobile && showMobileNav}">
        <li>
          <nuxt-link to="/">
            <nuxt-img src="~assets/images/j-icon.svg" alt="" />
          </nuxt-link>
        </li>
        <li v-for="navItem in navItems.find(x => x.text === 'More').submenu" :key="navItem.text">
          <nuxt-link v-if="navItem.url" :to="navItem.url">
            {{ navItem.text }}
          </nuxt-link>
        </li>
        <li role="presentation">
          <theme-switcher id="mobile" />
        </li>
      </ul>
    </div>
  </nav>
</template>

<script lang="ts">
import Vue from 'vue'
import ThemeSwitcher from '../ThemeSwitcher.vue'

interface ISubNav {
  text: String;
  url: String | null;
  icon?: String;
}

interface INav extends ISubNav {
  submenu: ISubNav[]
}

export default Vue.extend({
  name: 'Navigation',
  components: { ThemeSwitcher },
  data () {
    return {
      isMobile: false,
      isTouchscreen: false,
      showMobileNav: false,
      navItems: [
        {
          text: 'Home',
          url: '/',
          icon: 'home'
        },
        {
          text: 'About',
          url: '/about',
          icon: 'person'
        },
        {
          text: 'Projects',
          url: '/projects',
          icon: 'code'
        },
        {
          text: 'Blog',
          url: '/blog',
          icon: 'blog'
        },
        {
          text: 'More',
          url: null,
          submenu: [] as ISubNav[]
        }
      ] as INav[]
    }
  },
  watch: {
    // Close navigation menu on page change
    '$route' () {
      this.showMobileNav = false
    }
  },
  mounted (): void {
    this.setResponsiveness()
    window.addEventListener('resize', this.setResponsiveness)
  },
  destroyed (): void {
    window.removeEventListener('resize', this.setResponsiveness)
  },
  methods: {
    setResponsiveness (): void {
      const navBreak = window.getComputedStyle(document.querySelector('nav.nav')!).getPropertyValue('--nav-break')
      this.isMobile = !window.matchMedia(`(min-width: ${navBreak})`).matches
      this.isTouchscreen = !window.matchMedia('(hover: hover)').matches

    }
  }
})
</script>

<style lang="scss" scoped>
@use 'sass:math';

$nav-height: 4rem;

.nav {
  --nav-break: #{$responsive-small-desktop};
  position: fixed;
  z-index: 999;
  width: 100vw;
  background-color: var(--colour-foreground-1);

  @media (max-width: $responsive-small-desktop - math.div(1em, 16)) {
    bottom: 0;

    &__name,
    &__logo {
      display: none;
    }

    &__primary {
      position: relative;
      z-index: 2;
      box-shadow: var(--shadow-standard-reverse);
      background-color: var(--colour-foreground-1);

      &-list {
        display: flex;
        justify-content: center;
        padding-left: 0;
        margin: 0;
        list-style-type: none;
      }
    }

    &__item {
      width: 20%;

      &:nth-child(2) {
        margin-right: auto;
      }

      a {
        height: $nav-height;
        display: flex;
        flex-direction: column;
        align-items: center;
        justify-content: center;
        font-size: var(--text-small);
        text-decoration: none;
        text-align: center;
        color: var(--colour-text-secondary);
        will-change: color, background-color;
        transition-property: color, background-color;
        transition: var(--standard-animation-timing) ease;
      }

      svg {
        height: 2rem;
        width: 2rem;
      }
    }

    &__hamburger {
      cursor: pointer;
      display: flex;
      align-items: center;
      justify-content: center;
      position: absolute;
      bottom: 25%;
      left: 50%;
      transform: translateX(-50%);
      width: 3.375rem;
      height: 1.9375rem;
      background-color: var(--colour-primary);
      border: none;
      margin: 1rem 0;
      box-shadow: 0 0 8px rgba(0,0,0,0.4);

      &::before,
      &::after {
        content: "";
        position: absolute;
        z-index: 1;
        width: 2.375rem;
        height: 2.375rem;
        transform: scaleY(0.5774) rotate(-45deg);
        background-color: inherit;
        left: 0.5rem;
        box-shadow: 0 0 8px rgba(0,0,0,0.4);
      }

      &::before {
        top: -1.1875rem;
      }

      &::after {
        bottom: -1.1875rem;
      }

      > span {
        display: block;
        position: absolute;
        top: 0;
        left: 0;
        width: 3.375rem;
        height: 1.9375rem;
        z-index: 2;
        background: inherit;
      }

      &-icon {
        width: 1.3rem;
        height: 1.1rem;
        position: relative;
        z-index: 2;
        transform: rotate(0deg);
        cursor: pointer;

        span {
          display: block;
          position: absolute;
          height: 0.25rem;
          width: 100%;
          background-color: var(--colour-text-button);
          border-radius: 0.5rem;
          opacity: 1;
          left: 0;
          transform: rotate(0deg);
          transition: 0.25s ease-in-out;

          &:nth-child(1) {
            top: 0;
          }

          &:nth-child(2),
          &:nth-child(3) {
            top: 45%;
          }

          &:nth-child(4) {
            top: 90%;
          }

          .nav__hamburger--open & {
            &:nth-child(1),
            &:nth-child(4) {
              top: 45%;
              width: 0;
              left: 50%;
            }

            &:nth-child(2) {
              transform: rotate(45deg);
            }

            &:nth-child(3) {
              transform: rotate(-45deg);
            }
          }
        }
      }
    }

    &__more {
      position: fixed;
      z-index: 1;
      top: 100%;
      transition: top 0.4s ease;
      will-change: top;
      background-color: var(--colour-foreground-1);
      box-shadow: 0 0 8px 2px rgba(0,0,0,0.2);
      height: calc(100vh - #{$nav-height});
      overflow-y: auto;
      padding-left: 0;
      margin: 0;
      padding: 2rem 0;
      list-style-type: none;
      width: 100vw;

      &--mobile-open {
        top: 0;
      }

      img {
        width: 4rem;
        margin-inline: auto;
      }

      li {
        max-width: 100%;
        width: 100%;
        overflow: hidden;
        max-height: 3rem;
        transition: max-height 160ms ease;
        will-change: max-height;

        &:nth-last-child(2) {
          margin-bottom: 2rem;
        }

        &[role="presentation"] {
          max-height: unset;
          margin-top: 5rem;
        }
      }

      > li:first-of-type { // Logo
        max-height: unset;
        max-width: 18rem;
        margin-inline: auto;
        border-bottom-color: transparent;
      }

      a {
        text-decoration: none;
        display: flex;
        align-items: center;
        justify-content: space-between;
        padding: 0.8rem 1rem;
        position: relative;
        z-index: 12;
        color: var(--colour-text-primary);
      }
    }
  }

  @media (min-width: $responsive-small-desktop) {
    color: var(--colour-text-primary);
    z-index: 99;
    box-shadow: var(--shadow-standard);
    position: fixed;
    height: 4rem;

    &__wrapper {
      max-width: 104rem;
      margin-inline: auto;
      display: flex;
      align-items: stretch;
      gap: 1rem;
      justify-content: space-between;
      padding: 0.25rem 10rem;
    }

    &__name {
      color: var(--colour-text-primary) !important;
      font-weight: 700;
      font-size: var(--text-large);
      text-decoration: none;
      cursor: pointer;
      display: inline-flex;
      align-items: center;
    }

    &__logo {
      width: 3.5rem;
      position: relative;
      left: 10%;
    }

    &__primary {
      display: flex;

      &-list {
        list-style-type: none;
        padding-left: 0;
        margin: 0;
      }

      svg:not(.nav__item-chevron) {
        display: none;
      }
    }

    &__item {
      display: inline-flex;
      font-weight: 700;
      font-size: var(--text-large);
      height: 100%;

      &:hover,
      &:focus-within {
        ul {
          opacity: 1;
          visibility: visible;
        }
      }

      svg {
        height: 2rem;
        width: 2rem;
      }

      a {
        display: inline-flex;
        align-items: center;
        justify-content: center;
        padding: 0.25rem;
        height: 100%;
        width: 100%;
        min-width: 6.5rem; // 104px
        text-align: center;
        text-decoration: none;
        cursor: pointer;
        border-radius: 0.625rem;
        color: var(--colour-text-primary);
        will-change: color, background-color;
        transition-property: color, background-color;
        transition: var(--standard-animation-timing) ease;

        &:hover,
        &:focus {
          background-color: var(--colour-primary-faint);

          + ul {
            opacity: 1;
            visibility: visible;
          }
        }
      }

      ul {
        opacity: 0;
        visibility: hidden;
        will-change: opacity, visibility;
        transition-property: opacity, visibility;
        transition: var(--standard-animation-timing) ease;
        position: absolute;
        top: 100%;
        padding-left: 0;
        margin: 0;
        list-style-type: none;
        background-color: var(--colour-foreground-1);
        box-shadow: var(--shadow-standard);
        padding: 0.5rem 0.75rem;
        display: flex;
        flex-direction: column;
        align-items: center;
        gap: 0.25rem;
        font-size: var(--text-body);
        
        li {
          width: 100%;

          &:nth-last-child(2) {
            margin-bottom: 2rem;
          }
        }

        a {
          padding: 0.5rem;
        }
      }
    }

    &__hamburger {
      display: none;
    }

    &__more {
      display: none;
    }
  }

  a.nuxt-link-active {
    color: var(--colour-primary);
  }

  a.is-home:not(.nuxt-link-exact-active) {
    color: var(--colour-text-secondary);
  }
}
</style>