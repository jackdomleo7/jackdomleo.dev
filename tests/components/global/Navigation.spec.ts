import { mount, shallowMount } from '@vue/test-utils'
import { configureAxe, toHaveNoViolations } from 'jest-axe'
import Navigation from '@/components/global/Navigation.vue'

expect.extend(toHaveNoViolations)

const axe = configureAxe({
  rules: {
    region: { enabled: false } // Don't need to check because components are loaded in isolation
  }
})

describe('Navigation', () => {
  /**
   * Vue instance
   */

  it('is a Vue instance', () => {
    const wrapper = mount(Navigation, {
      attachTo: document.body,
      mocks: {
        $colorMode: {
          preference: 'system'
        }
      },
      stubs: ['nuxt-link', 'svg-icon', 'nuxt-img']
    })

    expect(wrapper.vm).toBeTruthy()

    wrapper.destroy()
  })

  /**
   * Snapshots
   */

  it('matches snapshot when mounted', () => {
    const wrapper = mount(Navigation, {
      attachTo: document.body,
      mocks: {
        $colorMode: {
          preference: 'system'
        }
      },
      stubs: ['nuxt-link', 'svg-icon', 'nuxt-img']
    })

    expect(wrapper.element).toMatchSnapshot()

    wrapper.destroy()
  })

  it('matches snapshot when shallow mounted', () => {
    const wrapper = shallowMount(Navigation, {
      attachTo: document.body,
      stubs: ['nuxt-link', 'svg-icon', 'nuxt-img']
    })

    expect(wrapper.element).toMatchSnapshot()

    wrapper.destroy()
  })

  /**
   * A11y
   */

  // ! FIX: For some reason, ul li[hidden="hidden"] is inaccessible ðŸ¤”
  it.skip('to have no obvious accessibility violations', async () => {
    const wrapper = mount(Navigation, {
      attachTo: document.body,
      mocks: {
        $colorMode: {
          preference: 'system'
        }
      },
      stubs: ['nuxt-link', 'svg-icon', 'nuxt-img']
    })

    await expect(axe(wrapper.element)).resolves.toHaveNoViolations()

    wrapper.destroy()
  })

  /**
   * Interactivity
   */

  it('shows the "More" nav list when the hamburger is pressed on mobile', async () => {
    Object.defineProperty(window, 'innerWidth', {
      writable: true,
      configurable: true,
      value: 375,
    })

    window.dispatchEvent(new Event('resize'))

    const wrapper = shallowMount(Navigation, {
      attachTo: document.body,
      stubs: ['nuxt-link', 'svg-icon', 'nuxt-img']
    })

    expect(wrapper.find('ul.nav__more').classes()).not.toContain('nav__more--mobile-open')

    await wrapper.find('button.nav__hamburger').trigger('click')

    expect(wrapper.find('ul.nav__more').classes()).toContain('nav__more--mobile-open')

    await wrapper.find('button.nav__hamburger').trigger('click')

    expect(wrapper.find('ul.nav__more').classes()).not.toContain('nav__more--mobile-open')

    wrapper.destroy()
  })
})
