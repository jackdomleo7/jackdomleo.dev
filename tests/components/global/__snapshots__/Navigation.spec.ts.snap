// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation matches snapshot when mounted 1`] = `
<nav
  class="nav"
>
  <div
    class="nav__wrapper"
  >
    <nuxt-link-stub
      class="nav__name"
      to="/"
    >
      
      Jack Domleo
      
      <span
        class="sr-only"
      >
         - Home
      </span>
    </nuxt-link-stub>
     
    <nuxt-img-stub
      alt=""
      class="nav__logo"
      src="~assets/images/j-icon.svg"
    />
     
    <div
      class="nav__primary"
    >
      <ul
        class="nav__primary-list"
      >
        <li
          class="nav__item"
          hidden="hidden"
        >
          <nuxt-link-stub
            to="/"
          >
            <svg-icon-stub
              name="home"
            />
             
            <span>
              Home
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <nuxt-link-stub
            to="/about"
          >
            <svg-icon-stub
              name="person"
            />
             
            <span>
              About
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <nuxt-link-stub
            to="/projects"
          >
            <svg-icon-stub
              name="code"
            />
             
            <span>
              Projects
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <nuxt-link-stub
            to="/blog"
          >
            <svg-icon-stub
              name="blog"
            />
             
            <span>
              Blog
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <a
            href="javascript:void(0);"
          >
            <span>
              More
            </span>
             
            <svg-icon-stub
              class="nav__item-chevron"
              name="chevron-down"
            />
          </a>
           
          <ul>
            <li>
              <nuxt-link-stub
                to="/contact"
              >
                
                Contact
              
              </nuxt-link-stub>
            </li>
            <li>
              <nuxt-link-stub
                to="/newsletter"
              >
                
                Newsletter
              
              </nuxt-link-stub>
            </li>
            <li>
              <nuxt-link-stub
                to="/speaking"
              >
                
                Speaking
              
              </nuxt-link-stub>
            </li>
             
            <li
              role="presentation"
            >
              <fieldset>
                <legend>
                  Switch theme
                </legend>
                 
                <div
                  class="theme-options"
                >
                  <div
                    class="theme-option"
                  >
                    <div>
                      <input
                        id="desktop-theme-system"
                        name="desktop-theme"
                        type="radio"
                        value="system"
                      />
                       
                      <svg-icon-stub
                        name="monitor"
                      />
                    </div>
                     
                    <label
                      for="desktop-theme-system"
                    >
                      System
                    </label>
                  </div>
                   
                  <div
                    class="theme-option"
                  >
                    <div>
                      <input
                        id="desktop-theme-light"
                        name="desktop-theme"
                        type="radio"
                        value="light"
                      />
                       
                      <svg-icon-stub
                        name="sun"
                      />
                    </div>
                     
                    <label
                      for="desktop-theme-light"
                    >
                      Light
                    </label>
                  </div>
                   
                  <div
                    class="theme-option"
                  >
                    <div>
                      <input
                        id="desktop-theme-dark"
                        name="desktop-theme"
                        type="radio"
                        value="dark"
                      />
                       
                      <svg-icon-stub
                        name="moon"
                      />
                    </div>
                     
                    <label
                      for="desktop-theme-dark"
                    >
                      Dark
                    </label>
                  </div>
                </div>
              </fieldset>
            </li>
          </ul>
        </li>
      </ul>
       
      <button
        class="nav__hamburger"
        style="display: none;"
      >
        <span />
         
        <div
          class="nav__hamburger-icon"
        >
          <span />
           
          <span />
           
          <span />
           
          <span />
        </div>
         
        <span
          class="sr-only"
        >
          Show less
        </span>
      </button>
    </div>
     
    <ul
      class="nav__more"
    >
      <li>
        <nuxt-link-stub
          to="/"
        >
          <nuxt-img-stub
            alt=""
            src="~assets/images/j-icon.svg"
          />
        </nuxt-link-stub>
      </li>
       
      <li>
        <nuxt-link-stub
          to="/contact"
        >
          
          Contact
        
        </nuxt-link-stub>
      </li>
      <li>
        <nuxt-link-stub
          to="/newsletter"
        >
          
          Newsletter
        
        </nuxt-link-stub>
      </li>
      <li>
        <nuxt-link-stub
          to="/speaking"
        >
          
          Speaking
        
        </nuxt-link-stub>
      </li>
       
      <li
        role="presentation"
      >
        <fieldset>
          <legend>
            Switch theme
          </legend>
           
          <div
            class="theme-options"
          >
            <div
              class="theme-option"
            >
              <div>
                <input
                  id="mobile-theme-system"
                  name="mobile-theme"
                  type="radio"
                  value="system"
                />
                 
                <svg-icon-stub
                  name="monitor"
                />
              </div>
               
              <label
                for="mobile-theme-system"
              >
                System
              </label>
            </div>
             
            <div
              class="theme-option"
            >
              <div>
                <input
                  id="mobile-theme-light"
                  name="mobile-theme"
                  type="radio"
                  value="light"
                />
                 
                <svg-icon-stub
                  name="sun"
                />
              </div>
               
              <label
                for="mobile-theme-light"
              >
                Light
              </label>
            </div>
             
            <div
              class="theme-option"
            >
              <div>
                <input
                  id="mobile-theme-dark"
                  name="mobile-theme"
                  type="radio"
                  value="dark"
                />
                 
                <svg-icon-stub
                  name="moon"
                />
              </div>
               
              <label
                for="mobile-theme-dark"
              >
                Dark
              </label>
            </div>
          </div>
        </fieldset>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Navigation matches snapshot when shallow mounted 1`] = `
<nav
  class="nav"
>
  <div
    class="nav__wrapper"
  >
    <nuxt-link-stub
      class="nav__name"
      to="/"
    >
      
      Jack Domleo
      
      <span
        class="sr-only"
      >
         - Home
      </span>
    </nuxt-link-stub>
     
    <nuxt-img-stub
      alt=""
      class="nav__logo"
      src="~assets/images/j-icon.svg"
    />
     
    <div
      class="nav__primary"
    >
      <ul
        class="nav__primary-list"
      >
        <li
          class="nav__item"
          hidden="hidden"
        >
          <nuxt-link-stub
            to="/"
          >
            <svg-icon-stub
              name="home"
            />
             
            <span>
              Home
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <nuxt-link-stub
            to="/about"
          >
            <svg-icon-stub
              name="person"
            />
             
            <span>
              About
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <nuxt-link-stub
            to="/projects"
          >
            <svg-icon-stub
              name="code"
            />
             
            <span>
              Projects
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <nuxt-link-stub
            to="/blog"
          >
            <svg-icon-stub
              name="blog"
            />
             
            <span>
              Blog
            </span>
          </nuxt-link-stub>
           
          <!---->
        </li>
        <li
          class="nav__item"
        >
          <a
            href="javascript:void(0);"
          >
            <span>
              More
            </span>
             
            <svg-icon-stub
              class="nav__item-chevron"
              name="chevron-down"
            />
          </a>
           
          <ul>
            <li>
              <nuxt-link-stub
                to="/contact"
              >
                
                Contact
              
              </nuxt-link-stub>
            </li>
            <li>
              <nuxt-link-stub
                to="/newsletter"
              >
                
                Newsletter
              
              </nuxt-link-stub>
            </li>
            <li>
              <nuxt-link-stub
                to="/speaking"
              >
                
                Speaking
              
              </nuxt-link-stub>
            </li>
             
            <li
              role="presentation"
            >
              <theme-switcher-stub
                id="desktop"
              />
            </li>
          </ul>
        </li>
      </ul>
       
      <button
        class="nav__hamburger"
        style="display: none;"
      >
        <span />
         
        <div
          class="nav__hamburger-icon"
        >
          <span />
           
          <span />
           
          <span />
           
          <span />
        </div>
         
        <span
          class="sr-only"
        >
          Show less
        </span>
      </button>
    </div>
     
    <ul
      class="nav__more"
    >
      <li>
        <nuxt-link-stub
          to="/"
        >
          <nuxt-img-stub
            alt=""
            src="~assets/images/j-icon.svg"
          />
        </nuxt-link-stub>
      </li>
       
      <li>
        <nuxt-link-stub
          to="/contact"
        >
          
          Contact
        
        </nuxt-link-stub>
      </li>
      <li>
        <nuxt-link-stub
          to="/newsletter"
        >
          
          Newsletter
        
        </nuxt-link-stub>
      </li>
      <li>
        <nuxt-link-stub
          to="/speaking"
        >
          
          Speaking
        
        </nuxt-link-stub>
      </li>
       
      <li
        role="presentation"
      >
        <theme-switcher-stub
          id="mobile"
        />
      </li>
    </ul>
  </div>
</nav>
`;
